python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/dbt_packages/test_columns_metadata.py

```diff
@@ -177,220 +215,255 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
         "trino",
     ],
 )
+@pytest.mark.parametrize(
+    "use_experimental_fetch_column_schema",
+    [True, False],
+)
 def test_column_lineage(
-    test_metadata_manifest: Dict[str, Any],
     sql_dialect: str,
+    test_metadata_manifest: Dict[str, Any],
     asset_key_selection: Optional[AssetKey],
+    use_experimental_fetch_column_schema: bool,
 ) -> None:
-    # Simulate the parsing of the SQL into a different dialect.
-    assert Dialect.get_or_raise(sql_dialect)
-
-    manifest = test_metadata_manifest.copy()
-    manifest["metadata"]["adapter_type"] = sql_dialect
+    with environ(
+        {"DBT_LOG_COLUMN_METADATA": str(not use_experimental_fetch_column_schema).lower()}
+    ):
```

Comment:
Instead of `environ`, use `monkeypatch.setenv` so we don't have to indent 200 lines